### PR TITLE
An Elm file should be considered compilable after initial error

### PR DESCRIPTION
**Pull request by icidasset**
👋 

I think this resolves the issue I talked about here https://github.com/hmsk/vite-plugin-elm/issues/9#issuecomment-852175564

This basically says that a file should still be considered as "compilable" even though an error occurs. So when `compiler.compileToString` fails, we'll still add it to the `compilableFiles` dictionary. This allows us to do HMR updates after a transform fails.

Will add some other notes as comments.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -407,17 +407,23 @@ export const plugin = (opts?: { debug: boolean }): Plugin => {
...
           output: '.js',
           optimize: !debug && isBuild,
           verbose: isBuild,
           debug,
         })
-        const dependencies: string[] = await compiler.findAllDependencies(id)
-        dependencies.forEach(this.addWatchFile)
-        compilableFiles.set(id, new Set(dependencies))
+
+        if (this.addWatchFile) {
```
Comment (icidasset) [inline src/index.ts:423]: I'm not sure why this is `undefined`

File: src/index.ts [hunk 2]
```diff
@@ -435,10 +441,8 @@ export const plugin = (opts?: { debug: boolean }): Plugin => {
           const message = `${viteProjectPath(
             id,
           )}: NO MAIN .elm file is requested to transform by vite. Probably, this file is just a depending module`
-          console.error(message)
           throw message
         } else {
-          console.error(e)
```
Comment (icidasset) [inline src/index.ts:441]: I removed these so we don't get double errors in the CLI. Was there a specific reason to do `console.error`?

Comment (hmsk) [reply src/index.ts:441]: unnecessary at all! thanks for pointing this out

File: src/index.ts (hunk 1 above)
Comment (hmsk) [reply src/index.ts:423]: hmm, how can we reproduce this situation. as long as I look through `addWatchFile` exists on context always

**Comment by hmsk** [comment]
Thanks for your contribution! This fix totally makes sense to me ✨

**Pull request merged**